.circleci/config.yml

```diff
@@ -3472,6 +3466,21 @@ workflows:
               max-dynamic-agents: [2]
           requires:
             - check-if-nightly-tests-should-run
+      - test-e2e-aws:
```

Comment:
I don't personally see much value in limiting the hugging-face context to only those tests that use it. But that's an opinion I have a hard time backing up, and it makes sense to me for other people to see it differently.

Re: resources, the change I see is to use `g4dn.12xlarge` instead of `g4dn.metal`. The AWS instance detail page tells me that `g4dn.12xlarge` is _smaller_ than `g4dn.metal`. I trust your intuition that these tests are more expensive than average. Does that mean we're over-provisioning? Do you think we should experiment with overall using the smaller `g4dn.12xlarge` instances?

I also think there's a real maintenance cost to right-sizing each test, but that intuition hasn't gelled yet. I'm up for a chat about it if you are. Maybe realtime.

I'm not asking for changes. I think this conversation is as much for me and to help me form an opinion of how I want our tests to be structured as it is for this specific decision.

Reply:
> The AWS instance detail page tells me that g4dn.12xlarge is smaller than g4dn.metal.

Oh, oops, you are right!  I was reading [this comment](https://github.com/determined-ai/determined/blob/6d4e9f039fe6e2c54dd202edb0598aa4d5b48a52/.circleci/config.yml#L3456-L3456) and just assumed that the `deepspeed` test resources (which I also use) were larger than the standard ones, but didn't actually check.  

> Does that mean we're over-provisioning?

I don't know, but I think the other cases which use `g4dn.metal` run many more tests, and so likely benefit from the larger allotment? More concurrent tests.   Because there are only two `huggingface` tests, it was easier to set the resources appropriately.

I think it might just make the most sense to add the `huggingface` context to the `test-e2e-gpu-distributed` tests. I'm not really sure of the utility of splitting out any of these distributed tests (like `test-e2e-gpu-transformers-amp`); they mostly all use the same resources as the top level `test-e2e-gpu-distributed`.

Reply:
>  I think the other cases which use g4dn.metal run many more tests, and so likely benefit from the larger allotment? More concurrent tests.

This would make a lot of sense, but I don't think it's the case. I'm pretty sure there's no parallelization right now in `test-e2e-gpu-distributed`, and... dang. It's really time for an audit.

Reply:
> I think it might just make the most sense to add the huggingface context to the test-e2e-gpu-distributed tests.

I don't feel much certainty in my own opinion, but that's the direction I lean, too.

Reply:
>I think it might just make the most sense to add the huggingface context to the test-e2e-gpu-distributed tests.

Did this.